src/main/java/core/basesyntax/service/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.impl.BalanceOperationService;
+import core.basesyntax.service.strategy.impl.PurchaseOperationService;
+import core.basesyntax.service.strategy.impl.ReturnOperationService;
+import core.basesyntax.service.strategy.impl.SupplyOperationService;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationService> map = Map.of(
+            FruitTransaction.Operation.BALANCE, new BalanceOperationService(),
+            FruitTransaction.Operation.SUPPLY, new SupplyOperationService(),
+            FruitTransaction.Operation.PURCHASE, new PurchaseOperationService(),
+            FruitTransaction.Operation.RETURN, new ReturnOperationService());
+
+    public void makeOperation(FruitTransaction fruitTransaction) {
+        OperationService operationService = map.get(fruitTransaction.getOperation());
+        operationService.calculateByOperation(fruitTransaction);
```

Comment:
let's method in this class just return an OperationService, and move calling perform operation to the class `SurplusCalculator`
```suggestion
    public OperationService getOperation(FruitTransaction fruitTransaction) {
        return map.get(fruitTransaction.getOperation());     
```